forem/forem
# Implement Unified Embeds for Forem-specific URLs

## What type of PR is this? (check all applicable)


## Description
This PR implements the [unified embed tag](https://github.com/forem/forem/issues/15099) for Forem-specific embeds.

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/15565
closes https://github.com/forem/forem/issues/15569
closes https://github.com/forem/forem/issues/15580
closes https://github.com/forem/forem/issues/15583
Note: The issues for LinkTag and OrganizationTag were already closed by former PRs

## QA Instructions, Screenshots, Recordings
For links to test with, grab the following link types from your `localhost`:
- Link to User Profile (old method: `{% user <username> %}`)
- Link to Organization Profile (old method: `{% org <org-slug> %}`)
- Link to Listing (old method: `{% listing <listing-url> %}`)
- Link to Podcast Episode (old method: `{% podcast <episode-url> %}`)
- Link to Article (old method: `{% link <article-url-or-slug %}`)
- Link to Tag page (old method: `{% tag <tag-name> %}`)

As a user, create Forem-specific liquid tag using this format: `{% embed <forem-specific-url> %}`. The Liquid Tag should render properly.
Also create liquid tags using the old method (examples above). This should also work, as I am leaving the old implementation in place for now.


### UI accessibility concerns?
None

## Added/updated tests?

## [Forem core team only] How will this change be communicated?
Considering the project as a whole, I shall update the Liquid Tag documentation at the end. Since I am not changing the current implementation, the present documentation is still valid. Plus, I will avoid updating the docs piecemeal, and risk confusing users.

## [optional] Are there any post deployment tasks we need to perform?
None
